docgram/README.md

```diff
@@ -0,0 +1,185 @@
+Grammar extraction tool for documentation
+=========================================
+
+doc_grammar extracts Coq's grammar from .mlg files, edits it and inserts it in
+chunks into .rst files.  The current prototype generates Sphinx
+`productionlist` displays.  We hope to also support `tacn`, `cmd`, `tacv` and
+`cmdv` displays soon.
+
+The mlg grammars present several challenges to generating an accurate grammar
+for documentation purposes:
+* The 30+ mlg files don't define an overall order in which nonterminals should
+appear in a complete grammar.
```

Comment:
That is not really an issue if grammar definitions are consolidated in consistent files, since we won't want to document the full grammar in a single Coq file anyway, but we would like to have a way to specify where each sub-part of the grammar should be inserted in the reference manual.

Reply:
It would be great if the mlg files could provide the overall order in a natural way.  But I didn't want to assume that was feasible, particularly since I don't understand everything about the parser.  It did seem like we use multiple mechanisms in the mlg files to do the same thing--would be good to be consistent.